src/simplify_algebra.cpp

```diff
@@ -1147,6 +1155,18 @@ struct find_splits
         })(ins1);
     }
 
+    /// Used to check if two module inputs have the same instructions.
+    static bool ins_eq(const module& x, const module& y)
+    {
+        auto get_mod_string = [](const module& m) {
+            std::stringstream ss;
+            m.print([&](auto ins, auto ins_names) { instruction::print(ss, ins, ins_names); });
+            return ss.str();
+        };
+        return get_mod_string(x) == get_mod_string(y);
+    }
```

Comment:
Why do you have this function instead of just doing `x == y`?

Reply:
For some reason it wasn't doing the equality correctly before. No idea what was the issue. Removing this func